projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/AbstractRouteBuilder.java

```diff
@@ -1,56 +1,70 @@
 package org.batfish.datamodel;
 
-public abstract class AbstractRouteBuilder<T extends AbstractRoute> {
+import org.batfish.common.BatfishException;
 
-  protected int _admin;
+public abstract class AbstractRouteBuilder<
+    S extends AbstractRouteBuilder<S, T>, T extends AbstractRoute> {
 
-  protected int _metric;
+  private int _admin;
 
-  protected Prefix _network;
+  private int _metric;
 
-  protected Ip _nextHopIp;
+  private Prefix _network;
 
-  protected int _tag;
+  private Ip _nextHopIp = Route.UNSET_ROUTE_NEXT_HOP_IP;
+
+  private int _tag = Route.UNSET_ROUTE_TAG;
 
   public abstract T build();
 
   public final Integer getAdmin() {
     return _admin;
   }
 
-  public final Integer getMetric() {
-    return _metric;
+  public final S setAdmin(int admin) {
+    _admin = admin;
+    return getThis();
   }
 
-  public final Prefix getNetwork() {
-    return _network;
+  // To handle the class casting exception while returning S in chaining methods
+  public abstract S getThis();
+
+  public final Integer getMetric() {
+    return _metric;
   }
 
-  public final Ip getNextHopIp() {
-    return _nextHopIp;
+  public final S setMetric(int metric) {
+    _metric = metric;
+    return getThis();
   }
 
-  public int getTag() {
-    return _tag;
+  public final Prefix getNetwork() {
+    return _network;
   }
 
-  public final void setAdmin(int admin) {
-    _admin = admin;
+  public final S setNetwork(Prefix network) {
+    if (network == null) {
+      throw new BatfishException("Cannot construct AbstractRoute with null network");
+    }
+    _network = network;
+    return getThis();
   }
 
-  public final void setMetric(int metric) {
-    _metric = metric;
+  public final Ip getNextHopIp() {
+    return _nextHopIp;
   }
 
-  public final void setNetwork(Prefix network) {
-    _network = network;
+  public final S setNextHopIp(Ip nextHopIp) {
+    _nextHopIp = nextHopIp == null ? Route.UNSET_ROUTE_NEXT_HOP_IP : nextHopIp;
```

Comment:
nextHopInterface in staticroute made consistent with this. Since Abstractroute does not declare  nextHopIp, I can think of to put this check only at this place